modules/garbage.py

```diff
@@ -2,124 +2,108 @@
-                print("At %.3f, GCI check the heap and it is at %.3f" % (self.env.now, server.heap.level))
-
-                if server.heap.level >= self.threshold:
-                    # create an event that will set shed as true and available as false
-                    self.shed_requests = True
-
-                    # wait for the queue to get empty.
-                    yield self.env.process(self.check_request_queue(server))
-
-                    if self.server.stwgc.collect_exe:
-                        self.server.stwgcc.collect_exe.interrupt()
+    def run_gc(self):
+        print("At %.3f, GCI check the heap and it is at %.3f" % (self.env.now, self.server.heap.level))
+        # create an event that will set shed as true and available as false
+        self.shed_requests = True
 
-                    # run STWGC
-                    gc_start_time = self.env.now
-                    yield self.env.process(server.stwgc.collect())
-                    gc_end_time = self.env.now
+        # wait for the queue to get empty.
+        yield self.env.process(self.check_request_queue())
 
-                    # leave server
-                    self.update_gci_values(gc_end_time - gc_start_time, server)
-                    self.shed_requests = False
+        # run GC
+        gc_start_time = self.env.now
+        if self.server.gc.collect_exe:
+            while self.server.gc.collect_exe:
+                yield self.env.timeout(self.sleep)
+        else:
+            yield self.env.process(self.server.gc.collect())
+        gc_end_time = self.env.now
 
-            yield self.env.timeout(self.sleep)
+        # leave server
+        self.update_gci_values(gc_end_time - gc_start_time)
+        self.shed_requests = False
+        self.times_performed += 1
+        print("At %.3f, GCI finish his job" % (self.env.now))
 
-    def check_request_queue(self, server):
-        while len(server.queue.items) > 0:
+    def check_request_queue(self):
+        while len(self.server.queue.items) > 0:
             yield self.env.timeout(self.sleep)
 
-    def estimated_shed_time(self, server):
-        return self.env.now + self.estimated_request_execution_time(server) + self.estimated_gc_execution_time(server)
+    def estimated_shed_time(self):
+        return self.estimated_request_execution_time() + self.estimated_gc_execution_time()
 
-    def estimated_request_execution_time(self, server):
+    def estimated_request_execution_time(self):
         # implement the way to get the max value of the last five time of requests execution time
         # history of lasts requests time executions
-        return self.gc_exec_time + len(server.queue.items) * 0.035
+        return self.gc_exec_time + len(self.server.queue.items) * 0.001
 
-    def estimated_gc_execution_time(self, server):
+    def estimated_gc_execution_time(self):
         # implement the way to get the max value of the last five time of requests execution time
```

Comment:
Is this a TODO? It should start with TODO(username): ....

Reply:
It should be. I'll fix it too. Thanks.

Reply:
Done.